### PR TITLE
Feature/forward iter

### PR DESCRIPTION
**Pull request by benjamin-work**

### CODE REVIEW

File: skorch/net.py [hunk 1]
```diff
@@ -612,8 +629,10 @@ def predict_proba(self, X):
         y_proba : numpy ndarray
 
         """
-        y_proba = self.forward(X, training=False)
-        y_proba = to_numpy(y_proba)
+        y_probas = []
+        for yp in self.forward_iter(X, training=False):
+            y_probas.append(to_numpy(yp))
```
Comment (ottonemo) [inline skorch/net.py]: Why not a list comprehension?

File: skorch/tests/test_net.py [hunk 2]
```diff
@@ -705,13 +705,13 @@ def test_target_1d_raises(self, net, data):
...
+    def test_predict_predict_proba(self, net_fit, data):
         X = data[0]
-        y_pred = net.predict(X)
+        y_pred = net_fit.predict(X)
 
         # predictions should not be all zeros
         assert not np.allclose(y_pred, 0)
 
-        y_proba = net.predict_proba(X)
+        y_proba = net_fit.predict_proba(X)
```
Comment (ottonemo) [inline skorch/tests/test_net.py]: Why this change and why didn't the previous version fail? This test doesn't do much if the unfitted net behaves the same?

Comment (benjamin-work) [reply skorch/tests/test_net.py]: The `net` fixture is uninitialized, which is why `predict` shouldn't work. Through side effects of other tests, this still works, but calling this test in isolation fails.

File: skorch/net.py (hunk 1 above)
Comment (benjamin-work) [reply skorch/net.py]: `y_probas = [to_numpy(yp) for yp in self.forward_iter(X, training=False)]` -> 80 chars with indents.

File: skorch/tests/test_net.py (hunk 2 above)
Comment (ottonemo) [reply skorch/tests/test_net.py]: And these side-effects are not a problem?
